app/models/mixins/has_monitoring_manager_mixin.rb

```diff
@@ -5,17 +5,15 @@ module HasMonitoringManagerMixin
 
   def ensure_monitoring_manager
     # monitoring_manager should be defined by child classes.
-    if try(:monitoring_manager_needed?)
+    if try(:monitoring_manager_needed?) && monitoring_manager.nil?
       build_monitoring_manager(:parent_manager => self)
       monitoring_manager.name = "#{name} Monitoring Manager"
-    end
-    ensure_monitoring_manager_properties
-  end
-
-  def ensure_monitoring_manager_properties
-    if monitoring_manager
       monitoring_manager.zone_id = zone_id
       monitoring_manager.provider_region = provider_region
     end
   end
+
+  def ensure_monitoring_manager_deletion
+    monitoring_manager.try(:delete_queue) unless try(:monitoring_manager_needed?)
+  end
```

Comment:
Theres a slight issue here. If a user tries to recreate the alerts endpoint while the previous manager wasnt deleted yet, he could end up with the endpoint but without a manager.

Reply:
maybe we could fail the save if `monitoring_manager.enabled = false` to reduce the chance of that?